qos/cometbft/store.go

```diff
@@ -0,0 +1,101 @@
+package cometbft
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+	"sync"
+
+	"github.com/pokt-network/poktroll/pkg/polylog"
+
+	"github.com/buildwithgrove/path/protocol"
+)
+
+// EndpointStore provides the endpoint selection capability required
+// by the protocol package for handling a service request.
+var _ protocol.EndpointSelector = &EndpointStore{}
+
+// EndpointStore maintains QoS data on the set of available endpoints
+// for an CometBFT-based blockchain service.å
+// It performs several tasks, most notable:
+//
+//	1- Endpoint selection based on the quality data available
+//	2- Application of endpoints' observations to update the data on endpoints.
+type EndpointStore struct {
+	Logger polylog.Logger
+
+	// ServiceState is the current perceived state of the CometBFT blockchain.
+	*ServiceState
+
+	endpointsMu sync.RWMutex
+	endpoints   map[protocol.EndpointAddr]endpoint
+}
+
+// Select returns an endpoint address matching an entry from the list of available endpoints.
+// available endpoints are filtered based on their validity first.
+// A random endpoint is then returned from the filtered list of valid endpoints.
+func (es *EndpointStore) Select(availableEndpoints []protocol.Endpoint) (protocol.EndpointAddr, error) {
```

Comment:
Ditto re a TODO on future refactoring (e.g. this method looks like a good candidate for a template method as part of an `EndpointStore` struct used across all QoS implementations. 

Such TODOs would help us focus on the PR's main objective, i.e. `CometBFT`  QoS, while leaving useful notes an ideas for future work.